pkg/kubernetes/nodes.go

```diff
@@ -131,3 +132,93 @@ func SetKubelet(mgr *manager.Manager, node *kubekeyapiv1alpha1.HostCfg) error {
+	if mgr.Runner.Index == 0 {
+		if out, err := mgr.Runner.ExecuteCmd("sudo -E /bin/sh -c \"set -o pipefail && /usr/local/bin/kubectl --kubeconfig /etc/kubernetes/admin.conf get configmap kube-proxy -n kube-system -o yaml "+
+			"| sed -n '/server:.*/p' \"", 1, false); err != nil {
+			return errors.Wrap(errors.WithStack(err), "Failed to get kube-proxy config")
+		} else {
+			if strings.Contains(strings.TrimSpace(out), LocalServer) {
+				return nil
+			}
+		}
+
+		if _, err := mgr.Runner.ExecuteCmd(fmt.Sprintf("sudo -E /bin/sh -c \"set -o pipefail && /usr/local/bin/kubectl --kubeconfig /etc/kubernetes/admin.conf get configmap kube-proxy -n kube-system -o yaml "+
+			"| sed 's#server:.*#server: https://127.0.0.1:%s#g' "+
+			"| /usr/local/bin/kubectl --kubeconfig /etc/kubernetes/admin.conf replace -f -\"", strconv.Itoa(mgr.Cluster.ControlPlaneEndpoint.Port)), 3, false); err != nil {
+			return errors.Wrap(errors.WithStack(err), "Failed to update kube-proxy config")
+		}
+	}
+
+	// Restart all kube-proxy pods to ensure that they load the new configmap.
+	if _, err := mgr.Runner.ExecuteCmd("sudo -E /bin/sh -c \"/usr/local/bin/kubectl --kubeconfig /etc/kubernetes/admin.conf delete pod -n kube-system -l k8s-app=kube-proxy --force --grace-period=0\"", 3, false); err != nil {
+		return errors.Wrap(errors.WithStack(err), "Failed to restart kube-proxy pod")
+	}
+	return nil
+}
+
+// UpdateKubectlConfig is used to update the kubectl config. Make the value of field 'server' to set as 127.0.0.1:6443 (default).
+// And all of the 'admin.conf' will connect to 127.0.0.1:6443 (default)
+func UpdateKubectlConfig(mgr *manager.Manager, node *kubekeyapiv1alpha1.HostCfg) error {
+	if node.IsMaster {
+		output, err := mgr.Runner.ExecuteCmd("sudo -E /bin/sh -c \"[ -f /etc/kubernetes/admin.conf ] && echo 'admin.conf is exists.' || echo 'admin.conf is not exists.'\"", 0, false)
+		if strings.Contains(output, "admin.conf is exists.") {
+			if _, err := mgr.Runner.ExecuteCmd(fmt.Sprintf("sudo sed -i 's#server:.*#server: https://%s:%s#g' /etc/kubernetes/admin.conf", node.InternalAddress, strconv.Itoa(mgr.Cluster.ControlPlaneEndpoint.Port)), 0, false); err != nil {
+				return errors.Wrap(errors.WithStack(err), "Failed to update /etc/kubernetes/kubelet.conf")
+			}
+		} else {
+			if err != nil {
+				return errors.Wrap(errors.WithStack(err), "Failed to find /etc/kubernetes/admin.conf")
+			}
+			return errors.New("Failed to find /etc/kubernetes/admin.conf")
+		}
+	}
+
+	output2, err := mgr.Runner.ExecuteCmd("sudo -E /bin/sh -c \"[ -f ~/.kube/config ] && echo 'kubectl config is exists.' || echo 'kubectl config is not exists.'\"", 0, false)
+	if strings.Contains(output2, "kubectl config is exists.") {
+		if _, err := mgr.Runner.ExecuteCmd(fmt.Sprintf("sudo sed -i 's#server:.*#server: https://127.0.0.1:%s#g' ~/.kube/config", strconv.Itoa(mgr.Cluster.ControlPlaneEndpoint.Port)), 0, false); err != nil {
```

Comment:
I think it is possible to reduce the `sed` operation by modifying the rule (`127.0.0.1  lb.kubesphere.local`) in `/etc/hosts`.